pkg/osquery/runtime/runner.go

```diff
@@ -306,6 +314,14 @@ func (r *Runner) launchOsqueryInstance() error {
 		return fmt.Errorf("fatal error starting osqueryd process: %w", err)
 	}
 
+	// wait for osquery to create the socket before moving on
+	if err := backoff.WaitFor(func() error {
+		_, err := os.Stat(paths.extensionSocketPath)
+		return err
+	}, 1*time.Minute, 1*time.Second); err != nil {
+		return fmt.Errorf("timeout waiting for osqueryd to create socket at %s: %w", paths.extensionSocketPath, err)
+	}
```

Comment:
Hrm.... You are extending the dragon region.

I think what this is trying to do is detect if osquery us running and healthy. (The comment should reflect that)

I naively think we should be using some `Healthy()` routine, and not just checking that the path exists. But if I go reading `Health()` I see that that we don't really separate the underlying health of osquery from the health of the extension manager. _And_ even if we did, we'd need to carefully work through socket contention and locks

So many checking for the socket is a simple thing we can do.

Otherwise, I think were looking at pulling in https://github.com/osquery/osquery-go/pull/107 and generally creating much more careful sequencing here.

Reply:
I think the dragon region is below a bit =), but yea it makes this func even longer.